ecosystem/typescript/sdk/src/tests/e2e/coin_client.test.ts

```diff
@@ -6,6 +6,7 @@ import { getFaucetClient, longTestTimeout, NODE_URL } from "../unit/test_helper.
 import { AptosAccount } from "../../account/aptos_account";
 import { COIN_TRANSFER, CoinClient, TRANSFER_COINS } from "../../plugins/coin_client";
 import { EntryFunctionPayload, Transaction_UserTransaction } from "../../generated";
+import { APTOS_COIN } from "../../utils";
 
 test(
   "transfer and checkBalance works",
```

Comment:
This test is gigantic and is chewing off more than it should. We should break this into several different tests:
1. Transferring and checking APT balances. This includes specifying vs not specifying coinType
2. Transferring and checking non-APT balances
3. Transferring and checking fungible assets balances
4. Error cases such as insufficient balances, invalid coin types, non-existent coin types/fungible assets

Reply:
This should also have at least a simple test for fungible asset transfer since it has that branch

Reply:
You can file an issue for this as well @xbtmatt